nearcore/src/cold_storage.rs

```diff
@@ -131,6 +207,27 @@ fn cold_store_loop(
     genesis_height: BlockHeight,
     runtime: Arc<NightshadeRuntime>,
 ) {
+    tracing::info!(target: "cold_store", "checking for initial migration");
+    loop {
+        if !keep_going.load(std::sync::atomic::Ordering::Relaxed) {
+            tracing::debug!(target: "cold_store", "stopping the initial migration loop");
+            break;
+        }
+        match cold_store_initial_migration(keep_going.clone(), &hot_store, cold_db.clone()) {
+            // We can either stop the cold store thread or hope that next time migration will not fail.
+            // Here we pick the second option.
+            Err(err) => {
+                tracing::error!(target: "cold_store", "initial migration failed with error {err}, sleeping 30s and trying again");
+                std::thread::sleep(std::time::Duration::from_secs(30));
+            }
+            // Any Ok status from `cold_store_initial_migration` function means that we can proceed to regular run.
+            Ok(status) => {
+                tracing::info!(target: "cold_store", "Initial migration status: {:?}. Moving on.", status);
+                break;
+            }
+        }
+    }
+
```

Comment:
Maybe move this to a separate method and call it before the cold store loop? For clarity and encaplusation I would rather keep the "initial migration" and "cold store loop" as separate things.

Reply:
Just to be clear: before the cold_store loop but still inside of the dedicated thread. Maybe something like this:
```
// in separate methods
spawn(move || {cold_store_initial_migration_and_loop(..)});

fn cold_store_initial_migration_and_loop(.. {
  cold_store_initial_migration()?;
  cold_store_loop()?;
}

// or even inline that in the spawn:
spawn(move || {
   cold_store_initial_migration()?;
   cold_store_loop()?;
}
```